bindings/cpp/src/Session.cpp

```diff
@@ -439,8 +454,26 @@ Subscribe::~Subscribe()
     }
 }
 
+void Subscribe::check_custom_loop_options(sr_subscr_options_t opts)
+{
+    if (uses_custom_loop && !(opts & SR_SUBSCR_NO_THREAD)) {
```

Comment:
let's check this the other way round as well -- if the ctor that created this was not for a custom event loop, error out if `SR_SUBSCR_NO_THREAD` is enabled

Reply:
done